src/Http/Utility.php

```diff
@@ -120,7 +120,7 @@ public function toAuthorizationHeader(array $params, $realm = null, $excludeCust
      */
     public function sign(
         array $params,
-        $signatureMethod,
+        string $signatureMethod,
```

Comment:
We can't really make a stricter signature here: it would be a BC break for child classes

Reply:
> no potential deprecation issues because of the non-strict type (since php 8.1 explode() now throws an error).

What deprecation warning This signature is entirely defined in this package :thinking: 

If this is about `null` being passed to `$signatureMethod`, we need a `(string)` cast in the body of the method, rather than an added type declaration.

Sadly, adding type declarations is considered a major BC breakage, so its "cost" is very high.

Reply:
Ah yes, I even voted on that RFC, but it was implemented in a terrible way :|